avm/res/container-service/managed-cluster/main.bicep

```diff
@@ -655,9 +658,12 @@ resource managedCluster 'Microsoft.ContainerService/managedClusters@2024-09-02-p
       omsagent: {
         enabled: omsAgentEnabled && !empty(monitoringWorkspaceResourceId)
         config: omsAgentEnabled && !empty(monitoringWorkspaceResourceId)
```

Comment:
To preserve linter / autocomplete support, a `shallowMerge()` might be better than `union()`
```bicep
        config: omsAgentEnabled && !empty(monitoringWorkspaceResourceId)
          ? {
              logAnalyticsWorkspaceResourceID: monitoringWorkspaceResourceId!
              ...(omsAgentUseAADAuth
                ? {
                    useAADAuth: 'true'
                  }
                : {})
            }
          : null
```

Reply:
Thanks. It is updated now.